src/vcProject.cpp

```diff
@@ -788,24 +790,54 @@ bool vcProject_UpdateNodeGeometryFromCartesian(vcProject *pProject, udProjectNod
     for (int i = 0; i < numPoints; ++i)
       pGeom[i] = udGeoZone_TransformPoint(pPoints[i], zone, pProject->baseZone);
 
-    result = udProjectNode_SetGeometry(pProject->pProject, pNode, newType, numPoints, (double*)pGeom);
+    for (int i = 0; i < numPoints * 3; ++i)
+    {
+      if (!std::isfinite(((double*)pGeom)[i]))
+      {
+        result = udE_InvalidParameter;
+        break;
+      }
+    }
+
+    if (result != udE_InvalidParameter)
+      result = udProjectNode_SetGeometry(pProject->pProject, pNode, newType, numPoints, (double*)pGeom);
```

Comment:
This will prevent the save happening (so bad data isn't written out) but bad data already in the file will still reproduce the crash. Not calling `udProjectNode_SetGeometry` will also not trigger `OnNodeUpdate` so the state of the node might be broken.